evo_researcher/benchmark/agents.py

```diff
@@ -103,34 +97,37 @@ def __init__(
         self.temperature = temperature
         self.embedding_model = embedding_model
 
-    def evaluate(self, market_question: str) -> EvaluatedQuestion:
-        return evaluate_question(question=market_question)
+    def is_predictable(self, market_question: str) -> bool:
+        return self.is_predictable_restricted(market_question, None)
 
-    def research(self, market_question: str) -> t.Optional[str]:
-        try:
-            return research_autonolas(
+    def is_predictable_restricted(self, market_question: str, time_restriction_up_to: datetime | None) -> bool:
+        return is_predictable(question=market_question)
```

Comment:
Looks like a circular loop of calls to `is_predictable` -> `is_predictable_restricted` -> `is_predictable` -> ...

Reply:
It's only confusing because this `is_predictable` is without `self`, so the loop isn't created, it calls a different function.